### DISCUSSION TITLE
Send location info when too low

### DISCUSSION THREAD

**Discussion by HanSolo** [NightscoutFoundation/xDrip#3221] [Ideas]
I wrote my own iOS app to monitor the glucose of our son and we found ourselves always checking the location of our son in case he is too low. He is using CamAPS FX with an Ypsopump. So I was wondering if there might be any chance to send the current location to an url in case the glucose is too low? In this case I could send his location to my server where I could for example send my app a push notification with the current location that can be opened in a map app. Thoughts?

**Comment by Navid200** [comment]
Is this related to xDrip?
xDrip can send an emergency message:
https://navid200.github.io/xDrip/docs/Emergency.html

**Comment by HanSolo** [comment]
Yep but that’s a short message, our problem is that if our son is outside somewhere and is too low, how do we find him in case it is urgent? It would be great if in this case xdrip+ could send a request with the gos coordinates to a url. Could be a simple json with a timestamp, the value and the lat/lon. With this info one could react on the situation with whatever is needed. Nightscout unfortunately doesn’t support location info.

**Reply by Navid200** [reply]
xDrip includes GPS coordinates, as well as a GPS pointer on Google map.

**Reply by HanSolo** [reply]
But how can I get the gos data in another app/server? Our son is using xdrip+ on his android phone (because of CamAPS FX) but we use iOS devices.

**Reply by Navid200** [reply]
If he is running xDrip on his device, xDrip will send a text message.  What does it matter what the receiving device is as long as it can receive a text message?
Does iOS not understand the GPS coordinates?  I don't use iOS.  But, GPS coordinates are not related to an OS.  They are universal.  Can you please explain what you mean?

**Reply by HanSolo** [reply]
Of course an iOs device can understand gos coordinates but this short message would only be send to one device. Sending a msg to an url would open the opportunity to subscribe with multiple devices e.g. me and my wife.

**Reply by Navid200** [reply]
xDrip can send the text message to multiple recipients.  Is that not enough?

What if you send something to a URL but, no one is monitoring the URL?  Why would that be better than a text message sent to specific contacts?

**Comment by HanSolo** [comment]
Well if it could send to multiple recipients it might work, let me give it a try. Oh and btw. an url call gives you way more opportunities than a text message.

**Reply by Navid200** [reply]
There are times we have to weigh the pros and cons and choose the simplest solution that accomplishes the main objective.
If an improvement will complicate the code and its maintenance, we may have no choice but to stay with the simple solution to allow xDrip to survive.  There are not many developers maintaining xDrip.  There is a lot of work.

At the end, there is always a judgement call that has to be made by the lead developer deciding if an addition/modification is worth the cost.  It's not an easy decision.
On one side, there are many users who ask for it.  On the other side, as soon as a feature is added, it becomes our responsibility to maintain it.

Right now, we have several features acting up after the upgrade to Android 14.  We are lucky that they are only a few features.  But, people are already complaining about them.  The maintenance part is not pretty.  And it must be done.  The simpler, the better for that reason.
